common/lib/transport/comettransport.ts

```diff
@@ -1,165 +1,177 @@
 import * as Utils from '../util/utils';
 import ProtocolMessage from '../types/protocolmessage';
-import Transport from '../transport/transport';
+import Transport from './transport';
 import Logger from '../util/logger';
 import Defaults from '../util/defaults';
 import ConnectionErrors from './connectionerrors';
 import Auth from '../client/auth';
 import ErrorInfo from '../types/errorinfo';
-
-var CometTransport = (function() {
-
-	var REQ_SEND = 0,
-		REQ_RECV = 1,
-		REQ_RECV_POLL = 2,
-		REQ_RECV_STREAM = 3;
-
-	/* TODO: can remove once realtime sends protocol message responses for comet errors */
-	function shouldBeErrorAction(err) {
-		var UNRESOLVABLE_ERROR_CODES = [80015, 80017, 80030];
-		if(err.code) {
-			if(Auth.isTokenErr(err)) return false;
-			if(Utils.arrIn(UNRESOLVABLE_ERROR_CODES, err.code)) return true;
-			return (err.code >= 40000 && err.code < 50000);
-		} else {
-			/* Likely a network or transport error of some kind. Certainly not fatal to the connection */
-			return false;
-		}
+import IXHRRequest from '../../types/IXHRRequest';
+import TokenDetails from '../../types/TokenDetails';
+
+type ConnectionManager = any;
+type TransportParams = any;
+
+const REQ_SEND = 0,
```

Comment:
I'm yet to learn TypeScript fully myself, but this looks like [an `enum`](https://www.typescriptlang.org/docs/handbook/enums.html). 🤔

Reply:
Good point, I've added an XHRStates enum in e9d0e80.